File: 567-permutation-in-string/567-permutation-in-string.cpp
class Solution {
public:
    bool checkInclusion(string s1, string s2) {
        if(s2.size()<s1.size())return 0;
        vector<int> x(26);
        vector<int> y(26);
        for(int i=0;i<s1.size();i++){
            x[s1[i]-'a']++;
            y[s2[i]-'a']++;
        }
        
        for(int i=0;i<=s2.size()-s1.size();i++){
            bool f=1;
            if(!i){
                for(int j=0;j<26;j++)if(x[j]!=y[j])f=0;
                if(f==1)return 1;
                continue;
            }
            y[s2[i-1]-'a']--;
            y[s2[i+s1.size()-1]-'a']++;
            for(int j=0;j<26;j++)if(x[j]!=y[j])f=0;
            if(f==1)return 1;
        }
        return 0;
    }
};

File: 567-permutation-in-string/567-permutation-in-string.py
class Solution:
    def checkInclusion(self, s1: str, s2: str) -> bool:
        l1=len(s1)
        l2=len(s2)
        if l2<l1:
            return 0
        x=[int(0)]*26
        y=[int(0)]*26
        # print(x)
        for i in range(l1):
            x[ord(s1[i])-ord('a')]+=1
        # print(x)
        for i in range(l2-l1+1):
            f=1
            if i==0:
                for j in range(l1):
                    y[ord(s2[j])-ord('a')]+=1
                if x==y:
                    return 1
                continue
            y[ord(s2[i-1])-ord('a')]-=1
            y[ord(s2[i+l1-1])-ord('a')]+=1
            
            if x==y:
                return 1
        # print(s)
        return 0